nav2_theta_star_planner/src/theta_star_planner.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <memory>
+#include <string>
+#include "nav2_theta_star_planner/theta_star_planner.hpp"
+#include "nav2_theta_star_planner/theta_star.hpp"
+
+namespace nav2_theta_star_planner
+{
+void ThetaStarPlanner::configure(
+  const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
+{
+  planner_ = std::make_unique<theta_star::ThetaStar>();
+  auto node = parent.lock();
+  logger_ = node->get_logger();
+  clock_ = node->get_clock();
+  name_ = name;
+  tf_ = tf;
+  planner_->costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".how_many_corners", rclcpp::ParameterValue(8));
+
+  node->get_parameter(name_ + ".how_many_corners", planner_->how_many_corners_);
+
+  if (planner_->how_many_corners_ != 8 && planner_->how_many_corners_ != 4) {
+    planner_->how_many_corners_ = 8;
+    RCLCPP_WARN(logger_, "Your value for - .how_many_corners  was overridden, and is now set to 8");
+  }
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".w_euc_cost", rclcpp::ParameterValue(4.0));
+  node->get_parameter(name_ + ".w_euc_cost", planner_->w_euc_cost_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".w_traversal_cost", rclcpp::ParameterValue(7.0));
+  node->get_parameter(name_ + ".w_traversal_cost", planner_->w_traversal_cost_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, name_ + ".w_heuristic_cost", rclcpp::ParameterValue(1.0));
+  node->get_parameter(name_ + ".w_heuristic_cost", planner_->w_heuristic_cost_);
+}
+
+void ThetaStarPlanner::cleanup()
+{
+  RCLCPP_INFO(logger_, "CleaningUp plugin %s of type nav2_theta_star_planner", name_.c_str());
```

Comment:
shouldn't this release the unique pointer to the planner? See the navfn planner for examples of what to do in these state transition functions